### PR TITLE
Elevator roller

### PR DESCRIPTION
**Pull request by Yotam7**

### CODE REVIEW

**Comment by drorta** [comment]
Fix this before people give you cr's

Tip: copy all of the files u wrote to somewhere, make sure they are saved, go to subsystems-dev, open a new branch with a different name and copy the files there

File: src/main/java/frc/robot/subsystems/elevatorRoller/ElevatorRoller.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		this.motor = elevatorRollerAvatiach.motor();
+		this.elevatorRollerDigitalInput = elevatorRollerAvatiach.digitalInput();
+		this.elevatorRollerAvatiach = elevatorRollerAvatiach;
+		this.elevatorRollerDigitalInputInputs = new DigitalInputInputsAutoLogged();
+
+		this.commandBuilder = new ElevatorRollerCommandBuilder(this);
+		updateInputs();
+	}
+
+	public ElevatorRollerCommandBuilder getCommandBuilder() {
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/elevatorRoller/ElevatorRoller.java:28]:
```suggestion
	public ElevatorRollerCommandsBuilder getCommandsBuilder() {
```

File: src/main/java/frc/robot/subsystems/elevatorRoller/ElevatorRoller.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		return commandBuilder;
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/elevatorRoller/ElevatorRoller.java:29]:
```suggestion
		return commandsBuilder;
```

File: src/main/java/frc/robot/subsystems/elevatorRoller/ElevatorRoller.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	public void updateInputs() {
+		elevatorRollerDigitalInput.updateInputs(elevatorRollerDigitalInputInputs);
+	}
+
+	public boolean isNoteIn() {
+		return elevatorRollerDigitalInputInputs.debouncedValue;
+	}
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/elevatorRoller/ElevatorRoller.java:38]: I don't think u have BB in elevator (it's connected to funnel motor)

File: src/main/java/frc/robot/subsystems/elevatorRoller/factory/RealElevatorRollerConstants.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+	private final static SparkLimitSwitch.Type REVERSE_LIMIT_SWITCH_TYPE = SparkLimitSwitch.Type.kNormallyOpen;
+	private final static Debouncer.DebounceType DEBOUNCE_TYPE = Debouncer.DebounceType.kBoth;
+	private final static double DEBOUNCE_TIME_SECONDS = 0.05;
+
+	public static ElevatorRollerAvatiach generate(String logPath) {
+		SparkMaxWrapper sparkMaxWrapper = new SparkMaxWrapper(IDs.ELEVATOR_ROLLER);
+
+		BrushedSparkMAXMotor brushedSparkMAXMotor = new BrushedSparkMAXMotor(logPath, sparkMaxWrapper);
+
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/elevatorRoller/factory/RealElevatorRollerConstants.java:23]: log voltage

**Review by YHerm — CHANGES_REQUESTED** [review]

**Pull request closed without merging**